vividus-soft-assert/src/test/java/org/vividus/softassert/model/SoftAssertionErrorTests.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.softassert.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.vividus.softassert.issue.KnownIssueType;
+
+class SoftAssertionErrorTests
+{
+    private final SoftAssertionError softAssertionError = new SoftAssertionError(new AssertionError());
+
+    @ParameterizedTest
+    @CsvSource({
+        "true,  false",
+        "false, true"
+    })
+    void testKnownIssue(boolean potentiallyKnown, boolean expectedResult)
+    {
+        KnownIssue knownIssue = new KnownIssue("Known issue", KnownIssueType.AUTOMATION, potentiallyKnown);
+        softAssertionError.setKnownIssue(knownIssue);
+        assertEquals(expectedResult, softAssertionError.isKnownIssue());
+        softAssertionError.setKnownIssue(null);
+        assertFalse(softAssertionError.isKnownIssue());
```

Comment:
unit test should assert only one business rule, but here at least 2 rules are validated and the last one is duplicated (one for each iteration)

Reply:
@valfirst 
can be a condition used here?
please review changes